tests/e2e/specs/product-editor/block-integration.test.js

```diff
@@ -0,0 +1,693 @@
+		await expect( editorUtils.getProductAttributesHeading() ).toBeVisible();
+
+		/*
+		 * 3 <select> for variable product:
+		 * - Channel visibility
+		 * - Brand (dynamically changed to google-listings-and-ads/product-select-with-text-field)
+		 * - Adult content
+		 */
+		await expect( panel.getByRole( 'combobox' ) ).toHaveCount( 3 );
+
+		/*
+		 * 0 <input type="text|date|time"> for variable product.
+		 */
+		await expect( panel.getByRole( 'textbox' ) ).toHaveCount( 0 );
+
+		/*
+		 * 0 <input type="number"> for variable product.
+		 */
+		await expect( panel.getByRole( 'spinbutton' ) ).toHaveCount( 0 );
+
+		// ===============================
+		// Go to edit the first variation.
+		// ===============================
+		await editorUtils.gotoEditVariationProductPage();
+		await editorUtils.clickPluginTab();
+
+		/*
+		 * 1 Section for variation product
+		 */
+		await expect( editorUtils.getProductAttributesHeading() ).toBeVisible();
+		await expect( editorUtils.getChannelVisibilityHeading() ).toHaveCount(
+			0
+		);
+
+		/*
+		 * 7 <select> for variation product:
+		 * - Condition
+		 * - Gender
+		 * - Size system
+		 * - Size type
+		 * - Age group
+		 * - Is bundle
+		 * - Adult content
+		 */
+		await expect( panel.getByRole( 'combobox' ) ).toHaveCount( 7 );
+
+		/*
+		 * 8 <input type="text|date|time"> for variation product:
+		 * - GTIN
+		 * - MPN
+		 * - Size
+		 * - Color
+		 * - Material
+		 * - Pattern
+		 * - Availability date
+		 * - Availability time
+		 */
+		await expect( panel.getByRole( 'textbox' ) ).toHaveCount( 8 );
+
+		/*
+		 * 1 <input type="number"> for variation product:
+		 * - Multipack
+		 */
+		await expect( panel.getByRole( 'spinbutton' ) ).toHaveCount( 1 );
+	} );
+
+	test( 'Channel visibility is disabled when hiding in product catalog', async () => {
+		await editorUtils.gotoAddProductPage();
+		await editorUtils.fillProductName();
+
+		const { selection, help } = editorUtils.getChannelVisibility();
+
+		await editorUtils.clickPluginTab();
+		await expect( selection ).toBeEnabled();
+		await expect( selection ).toHaveValue( 'sync-and-show' );
+		await expect( help ).toHaveCount( 0 );
+
+		await editorUtils.clickTab( 'Organization' );
+		await page.getByLabel( 'Hide in product catalog' ).setChecked( true );
+		await editorUtils.save();
+
+		await editorUtils.clickPluginTab();
+		await expect( selection ).toBeDisabled();
+		await expect( selection ).toHaveValue( 'dont-sync-and-show' );
+		await expect( help ).toBeVisible();
+		await expect( help ).toContainText(
+			'This product cannot be shown on any channel because it is hidden from your store catalog.'
+		);
+
+		await editorUtils.clickTab( 'Organization' );
+		await page.getByLabel( 'Hide in product catalog' ).setChecked( false );
+		await editorUtils.save();
+
+		await editorUtils.clickPluginTab();
+		await expect( selection ).toBeEnabled();
+		await expect( help ).toHaveCount( 0 );
```

Comment:
Why are we testing again this part? Maybe we can reduce execution time as 'Hide in product catalog' is false by default and is tested already.

Reply:
This asserts that the channel visibility status can be changed back from disabled to enabled.